File: TestApp/app/src/main/cpp/TestClass.cpp
#include "TestClass.hpp"

#include <Hiena/utility/ImplementationMacros.hpp>

namespace com::testapp::example
{
	TestClass::TestClass(hiena::CreateNew_t)
		:TestClass(hiena::NewObject<TestClass>())
	{}

	TestClass::TestClass(java::lang::String Text)
		:TestClass(hiena::NewObject<TestClass>(Text))
	{}
}

File: TestApp/app/src/main/java/com/testapp/example/TestClass.java
package com.testapp.example;

public class TestClass {
    public int A;
    public float B;

    static String S = new String("Hello from Java side");

    TestClass()
    {
        A = 1;
        B = 2;
    }

    TestClass(String Text)
    {
        A = 3;
        B = 4;
        S = Text;
    }
}